jerry-core/vm/vm.c

```diff
@@ -794,92 +769,70 @@ vm_loop (vm_frame_ctx_t *frame_ctx_p) /**< frame context */
             break;
           }
         }
-        if (CBC_BRANCH_IS_BACKWARD (opcode_flags))
+
+        if (opcode_data & VM_OC_BACKWARD_BRANCH)
         {
           branch_offset = -branch_offset;
         }
       }
 
-      free_flags = 0;
+      left_value = ecma_make_simple_value (ECMA_SIMPLE_VALUE_UNDEFINED);
+      right_value = ecma_make_simple_value (ECMA_SIMPLE_VALUE_UNDEFINED);
+
       if (VM_OC_HAS_GET_ARGS (opcode_data))
       {
-        uint32_t operands = VM_OC_GET_ARGS_GET_INDEX (opcode_data);
+        uint32_t operands = VM_OC_GET_ARGS_INDEX (opcode_data);
 
-        if (operands >= VM_OC_GET_ARGS_GET_INDEX (VM_OC_GET_LITERAL))
+        if (operands >= VM_OC_GET_LITERAL)
         {
           uint16_t literal_index;
           READ_LITERAL_INDEX (literal_index);
           READ_LITERAL (literal_index,
-                        left_value,
-                        free_flags = VM_FREE_LEFT_VALUE);
+                        left_value);
 
           switch (operands)
           {
-            case VM_OC_GET_ARGS_GET_INDEX (VM_OC_GET_STACK_LITERAL):
-            {
-              JERRY_ASSERT (stack_top_p > frame_ctx_p->registers_p + register_end);
-              right_value = left_value;
-              left_value = *(--stack_top_p);
-              free_flags = (uint8_t) ((free_flags << 1) | VM_FREE_LEFT_VALUE);
-              break;
-            }
-            case VM_OC_GET_ARGS_GET_INDEX (VM_OC_GET_LITERAL_BYTE):
+            case VM_OC_GET_LITERAL:
```

Comment:
Yes. The VM_OC_GET_LITERAL has the highest chance, so it would be great if that would be checked first. The original "if" statement did it as well.